src/color_calibration/coco_integration.py

```diff
@@ -0,0 +1,150 @@
+import cv2
+from pathlib import Path
+from .coco_utils import (
+    load_coco_annotations,
+    get_color_card_annotation,
+    extract_color_card
+)
+from .match_histograms import match_histograms
+
+class DatasetPaths:
+    """Helper class to manage dataset paths and structure"""
+    def __init__(self, dataset_base_dir, dataset_output_dir):
+        self.base_dir = Path(dataset_base_dir)
+        self.output_dir = Path(dataset_output_dir)
+        self.resized_dir = self.output_dir / "resized"
+        
+        if not self.base_dir.exists():
+            raise FileNotFoundError(f"Dataset base directory not found: {self.base_dir}")
+        if not self.output_dir.exists():
+            raise FileNotFoundError(f"Dataset output directory not found: {self.output_dir}")
```

Comment:
```suggestion
        self.base_dir = Path(dataset_base_dir)
        if not self.base_dir.exists():
            raise FileNotFoundError(f"Dataset base directory not found: {self.base_dir}")
        self.output_dir = Path(dataset_output_dir)
        if not self.output_dir.exists():
            raise FileNotFoundError(f"Dataset output directory not found: {self.output_dir}")
        
        self.resized_dir = self.output_dir / "resized"
```
Should save the definition of resized directory to after confirmation that the output directory exists.